src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -34,19 +34,19 @@ public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, b
             if (outputBuffer == null)
                 ThrowHelper.ThrowArgumentNull(ThrowHelper.ExceptionArgument.outputBuffer);
 
-            // For now, only convert 3 bytes to 4
-            Span<byte> input = inputBuffer.AsSpan(inputOffset, InputBlockSize);
-            Span<byte> output = outputBuffer.AsSpan(outputOffset, OutputBlockSize);
+            int inputBlocks = inputCount / InputBlockSize;
+            int outputBlocks = (outputBuffer.Length - outputOffset) / OutputBlockSize;
+            int blocksToTransform = Math.Min(inputBlocks, outputBlocks);
+            Debug.Assert(blocksToTransform >= 1);
 
-            OperationStatus status = Base64.EncodeToUtf8(input, output, out int consumed, out int written, isFinalBlock: false);
+            Span<byte> input = inputBuffer.AsSpan(inputOffset, blocksToTransform * InputBlockSize);
+            Span<byte> output = outputBuffer.AsSpan(outputOffset, blocksToTransform * OutputBlockSize);
 
-            if (written != OutputBlockSize)
-            {
-                ThrowHelper.ThrowCryptographicException();
-            }
+            OperationStatus status = Base64.EncodeToUtf8(input, output, out int consumed, out int written, isFinalBlock: false);
 
             Debug.Assert(status == OperationStatus.Done);
-            Debug.Assert(consumed == InputBlockSize);
+            Debug.Assert(consumed == input.Length);
+            Debug.Assert(written == output.Length);
 
             return written;
         }
```

Comment:
Can't reach low enough:  TransformFinalBlock also needs to be updated to support multiple blocks.

Reply:
Ah, ok.  This contract is... interesting.